cmd/codeqlExecuteScan.go

```diff
@@ -185,13 +187,50 @@ func uploadResults(config *codeqlExecuteScanOptions, repoInfo RepoInfo, token st
 
 	//if no git pramas are passed(commitId, reference, serverUrl, repository), then codeql tries to auto populate it based on git information of the checkout repository.
 	//It also depends on the orchestrator. Some orchestrator keep git information and some not.
+
+	var buffer bytes.Buffer
+	utils.Stdout(&buffer)
 	err := execute(utils, cmd, GeneralConfig.Verbose)
 	if err != nil {
 		log.Entry().Error("failed to upload sarif results")
-		return err
+		return "", err
 	}
+	utils.Stdout(log.Writer())
+
+	url := buffer.String()
+	return strings.TrimSpace(url), nil
+}
 
-	return nil
+func waitSarifUploaded(codeqlSarifUploader codeql.CodeqlSarifUploader) error {
+	var sarifUploadComplete = "complete"
+	var sarifUploadFailed = "failed"
+	var maxRetries = 5
```

Comment:
we should make the timeout configurable. it can be a separate PR, though. as you like.

Reply:
added configurable parameters